### PR TITLE
configure: remove coqide switch

### PR DESCRIPTION
**Pull request by SkySkimmer**
It was basically a noop since the move to dune.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
Hidden merge conflict with https://github.com/coq/coq/pull/16505, the one merged later will need fixing.

**Comment by Blaisorblade** [comment]
The fix is to stop passing `-coqide` at all? That explains why `-coqide opt` and passing nothing were equivalent; one could drop that from #16505.

**Comment by Zimmi48** [comment]
Instead of removing it directly, could you keep it as a no-op with a deprecation warning? We've done this before for the removal of other no-op options and this will help users understand what has happened. Maybe add a changelog entry too.

**Comment by SkySkimmer** [comment]
IMO deprecations in configure are invisible and worse than useless (as it appears that the switch is working but actually does nothing).
A changelog is definitely needed though.

**Comment by Zimmi48** [comment]
>IMO deprecations in configure are invisible and worse than useless (as it appears that the switch is working but actually does nothing).

OK, not a deprecation then, an error message when the option is used. But also fine to just go with a changelog.

**Comment by Zimmi48** [comment]
still broken...

File: doc/changelog/12-infrastructure-and-dependencies/16512-conf-nociqodie.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- **Removed:**
+  the ``-coqide`` switch to ``configure`` in Coq's build infrastructure
+  (it stopped controlling what got compiled in the move to dune)
```
Comment (Alizter) [inline doc/changelog/12-infrastructure-and-dependencies/16512-conf-nociqodie.rst:3]:
```suggestion
  (it stopped controlling what got compiled in the move to Dune)
```

**Review by Alizter — APPROVED** [review]

**Review by ejgallego — COMMENTED** [review]
Note that this will remove the lablgtk check that some users rely for. An alternative (and what I was planning to do) is to introduce coqide/configure

For sure, `gtk_platform` should live in `ide/coqide/config` not in `config/coq_config`.

**Comment by SkySkimmer** [comment]
>that some users rely for

Who and what for?

**Comment by ejgallego** [comment]
> Who and what for?

Maybe Jason, I don't remember. I'm fine with this, but note people maybe still relying on this.

**Comment by Alizter** [comment]
Doesn't Dune check for lablgtk now?

**Comment by ejgallego** [comment]
> Doesn't Dune check for lablgtk now?

Dune check on build, not on configure. Anyways I think this OR goes on the right direction IMHO, cc #7149

**Comment by Alizter** [comment]
So what do we do? Go with the new configure or this. IMO this seems better.

**Comment by Alizter** [comment]
@ejgallego your coqide configure https://github.com/coq/coq/pull/16531

**Comment by ejgallego** [comment]
> @ejgallego your coqide configure #16531

That PR is orthogonal to this one.

**Comment by RalfJung** [comment]
I think this PR broke our CI, see https://github.com/coq/coq/issues/16548

**Pull request merged**